subsys/net/l2_wifi_if_conn/l2_wifi_conn.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(l2_wifi_mgr_conn);
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/net_event.h>
+#include <net/wifi_mgmt_ext.h>
+#include <net/l2_wifi_connect.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include <zephyr/src/supp_events.h>
+#include "conn_mgr_private.h"
+
+#define WIFI_SHELL_MGMT_EVENTS (NET_EVENT_WIFI_CONNECT_RESULT | \
+				NET_EVENT_WIFI_DISCONNECT_RESULT)
+
+#define WPA_SUPP_EVENTS (NET_EVENT_WPA_SUPP_READY)
+
+static struct net_mgmt_event_callback net_l2_mgmt_cb;
+static struct net_mgmt_event_callback net_wpa_supp_cb;
+int connection_status;
+
+int net_l2_wifi_connect(struct conn_mgr_conn_binding *const binding);
+int net_l2_wifi_disconnect(struct conn_mgr_conn_binding *const binding);
+
+K_SEM_DEFINE(wpa_supp_ready_sem, 0, 1);
+
+/* Delayable work used to handle Wi-Fi connection timeouts. */
+static void wifi_conn_work_handler(struct k_work *work);
+K_THREAD_STACK_DEFINE(l2_wifi_wq_stack_area, CONFIG_L2_WIFI_CONN_WQ_STACK_SIZE);
+struct k_work_q wifi_conn_wq;
+static K_WORK_DELAYABLE_DEFINE(wifi_conn_work, wifi_conn_work_handler);
+
+/* Store a local reference to the connection binding */
+static struct net_if *wifi_iface;
+
+static void wifi_conn_work_handler(struct k_work *work)
+{
+	if (connection_status != WIFI_STATE_COMPLETED) {
+		net_l2_wifi_disconnect(NULL);
+	}
+}
+
+static int wifi_conn_timeout_schedule(void)
+{
+	int rc;
+
+	rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, wifi_iface, NULL, 0);
```

Comment:
I would move this line into `net_l2_wifi_connect` just before the call to `wifi_conn_timeout_schedule`, just to help the function naming line up better with function behavior

Reply:
Done